uplink/piecestore/download.go

```diff
@@ -36,14 +36,14 @@ type Download struct {
 }
 
 func (client *Client) Download(ctx context.Context, limit *pb.OrderLimit2, offset, size int64) (*Download, error) {
-	peer, err := identity.PeerIdentityFromContext(ctx)
+	stream, err := client.client.Download(ctx)
```

Comment:
I wonder whether we can get peer identity from *grpc.ClientConn. I'm just trying to avoid sending the orderlimit unless we already have peer identity.

Reply:
but we are sending OrderLimit few lines later. We are getting stream, peer identity and then we are using stream

Reply:
ah, yeah, nevermind

Reply:
maybe name `Upload` for both methods is confusing. We have `Upload` in `Upload`

Reply:
Yeah, I think so, but using different names might be confusing as well. We can put into bikeshed if you wish.